### ISSUE TITLE
OpenVR-AdvancedSettings only works when HMD plugged into the "wrong" graphics card?

### ISSUE THREAD

**Issue by guiglass** [OpenVR-Advanced-Settings/OpenVR-AdvancedSettings#585]
SteamVR 1.21.8 with OpenVr settings 5.3.2

Used to work a few weeks ago but I have tried so many times to uninstall everything and reinstall, but each time I start SteamVR and click OVR icon the panel is completely transparent.

Except if I plug the HMD into the "wrong" graphics card then the panel works fine but I can't play any games because SteamVR says "Wrong Graphics Card" and the game is no longer in VR.

Has anything changed in SteamVR? I'm not sure what to do but I can't switch gpu every time I need to change a setting. Is there a way to fix this?

**Comment by Myrkie** [comment]
Hi, I believe I know the cause of this issue, this is more of a limitation based on SteamVR itself, by default SteamVR will try to render on whatever GPU its connected to, usually your dedicated graphics card such as an NVIDIA Card, but if you happen to use a system that also contains an IGPU or uses NVIDIA optimus, SteamVR will still render on the second GPU because of power control in windows but any external tools will run on the first GPU it finds by index usually being the IGPU, since the SteamVR render stack is rendering on the dedicated GPU while any overlay is not this can make the interface appear invisible, this can be easily fixed for unity based overlays or applications with the launch option `-force-device-index INT`for non unity based overlays they will have to add the support themselves.

**Comment by ykeara** [comment]
so ive done a pass at this but its a opengl issue as far as i can tell and options are limited in nvidia setting forcing advanced settings to run on gpu may fix issue

**Comment by Myrkie** [comment]
> so ive done a pass at this but its a opengl issue as far as i can tell and options are limited in nvidia setting forcing advanced settings to run on gpu may fix issue

yeah from my testing, multiGPU or having integrated graphics is not very well supported in windows. NVIDIA settings are garbage and usually don't work, Microsoft did add a method to change which GPU a program runs on though this is ONLY for windows 11, it was in preview builds for 10 but was later dropped as a feature. if your having issues forcing it to use your dedicated GPU through the graphics settings in windows 10 then your out of luck.

![image](https://user-images.githubusercontent.com/20288698/221364131-32c6adc6-984b-4c47-99d0-91c09ee2ced4.png)
